File: src/main/java/com/bazbatlabs/smashballs/controllers/MenuController.java
package com.bazbatlabs.smashballs.controllers;

import android.view.KeyEvent;

import tv.ouya.console.api.OuyaController;

import com.bazbatlabs.smashballs.models.*;
import com.bazbatlabs.smashballs.views.MenuView;

public final class MenuController implements Controller {

    private final Menu menu;
    private final MenuView view;
    private final Artist artist;
    private final WorldController worldController;

    public MenuController(Artist artist, ImageMap images, SoundMap sounds, WorldController worldController) {
        this.menu = new Menu();
        this.view = new MenuView(this.menu, images, sounds, artist);
        this.artist = artist;
        this.worldController = worldController;
    }

    @Override
    public void draw(int screenWidth, int screenHeight) {
        view.draw();
    }

    @Override
    public Controller update() {
        menu.update();

        if (menu.isReadyToStartGame()) {
            return worldController;
        } else {
            return this;
        }
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        boolean handled = true;

        switch (keyCode) {
        case OuyaController.BUTTON_DPAD_UP:
            menu.previousEntry();
            break;

        case OuyaController.BUTTON_DPAD_DOWN:
            menu.nextEntry();
            break;

        case OuyaController.BUTTON_O:
            menu.select();
            break;

        default:
            handled = false;
            break;
        }

        return handled;
    }

    @Override
    public boolean onKeyUp(int keyCode, KeyEvent event) {
        return false;
    }

    @Override
    public void changeSurface(int width, int height) {
        artist.changeSurface(width, height);
    }
}

File: src/main/scala/controllers/MenuController.scala
package com.bazbatlabs.bricksmash.controllers

import android.view.KeyEvent
import android.content.res.Resources

import tv.ouya.console.api.OuyaController

import com.bazbatlabs.bricksmash.lib._
import com.bazbatlabs.bricksmash.models._
import com.bazbatlabs.bricksmash.views._
import com.bazbatlabs.bricksmash.views.MenuView

class MenuController(val artist: Artist, val images: ImageMap,
                     val sounds: SoundMap, val resources: Resources) extends Controller {

  val menu = new Menu()
  val view = new MenuView(menu, images, sounds, artist)

  override def draw(screenWidth: Int, screenHeight: Int) { view.draw() }

  override def update(): Controller = {
    menu.update()

    if (menu.isReadyToStartGame) {
      new WorldController(artist, images, sounds, resources)
    } else {
      this
    }
  }

  override def onKeyDown(keyCode: Int, event: KeyEvent): Boolean = {
    var handled = true

    keyCode match {
      case OuyaController.BUTTON_DPAD_UP => menu.previousEntry()
      case OuyaController.BUTTON_DPAD_DOWN => menu.nextEntry()
      case OuyaController.BUTTON_O => menu.select()
      case _ => handled = false
    }

    handled
  }

  override def onKeyUp(keyCode: Int, event: KeyEvent) = false

  override def changeSurface(width: Int, height: Int) {
    artist.changeSurface(width, height)
  }
}
